### PR TITLE
[bitnami/kong] MAJOR - Kong subchart bump

### PR DESCRIPTION
**Pull request by joancafom**
**Description of the change**

This PR bumps Kong major version since its dependencies (both PostgreSQL and Cassandra) bumped their major versions including breaking changes. More info at #3021 (PostgreSQL) and (Cassandra) #3805.

It also introduces some common standardizations found across charts, such as affinityPressets and common labels & annotations.

**Benefits**

Up-to-date subcomponents, standardization 

**Possible drawbacks**

Breaks backwards compatibility

**Applicable issues**
  NA

**Additional information**
This PR uses some of the validations introduced in #4110 , should merge those changes and update `requirements.lock` before merging this one

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/kong/README.md [hunk 1]
```diff
@@ -410,3 +417,9 @@ $ helm upgrade my-release bitnami/kong \
 ```
 
 > Note: you need to substitute the placeholders _[POSTGRESQL_PASSWORD]_ with the values obtained from instructions in the installation notes.
+
+### To 2.0.0
+
+PostgreSQL and Cassandra dependency versions were bumped to new major versions, `9.x.x` and `6.x.x` respectively. Both of these include breaking changes and hence backwards compatibility is no longer guaranteed.
+
+Depending on your backend, you may refer to the [PostgreSQL Upgrade Notes](https://github.com/bitnami/charts/tree/master/bitnami/postgresql#900) or the [Cassandra Upgrade Notes](https://github.com/bitnami/charts/tree/master/bitnami/cassandra#to-600) to properly migrate your data to this new version.
```
Comment (juan131) [inline bitnami/kong/README.md:425]: > Depending on your backend

What do you mean? Please elaborate it properly.

---

Does Kong provides its own way to bump/restore the data?

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the PR!! Please take a look to my comments

File: bitnami/kong/requirements.lock [hunk 2]
```diff
@@ -5,5 +5,8 @@ dependencies:
 - name: cassandra
   repository: https://charts.bitnami.com/bitnami
   version: 5.6.7
-digest: sha256:59edb19078fb43fd0caa5dc4951c48fa40fb09d3bfea19c0f6556070b6f9d875
-generated: "2020-09-24T00:29:07.539774133Z"
+- name: common
+  repository: https://charts.bitnami.com/bitnami
+  version: 0.9.0
+digest: sha256:a9f7cc97b07b9400dcecfe961f4330193c27b818a876ecdf0478747be63981e6
+generated: "2020-10-22T22:10:49.045717+02:00"
```
[the remark this answers is not in the document]
Comment (joancafom) [reply bitnami/kong/requirements.lock:12]: This PR uses some of the validations introduced in #4110 so I was waiting until merging those ones in order to update `requirements.lock`. Thanks for pointing it out though 😄

File: bitnami/kong/README.md (hunk 1 above)
Comment (joancafom) [reply bitnami/kong/README.md:425]: Kong could be integrated with either PostgreSQL or Cassandra as per user preference. The property `database` in values lets you choose between this two. I will update the README to make this more clear 👍.

> Does Kong provides its own way to bump/restore the data?

To the best of my knowledge, it does not provide with its own feature to restore backup/restore the data. As our Kong setup relies on DBs to store configuration, restoring them would be sufficient.

File: bitnami/kong/requirements.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/kong/requirements.lock:12]: In this case, I'd block this PR until then

**Comment by joancafom** [comment]
Changes implemented @juan131

File: bitnami/kong/README.md [hunk 3]
```diff
@@ -80,6 +82,11 @@ The following tables list the configurable parameters of the kong chart and thei
...
+| `nodeAffinityPreset.type`                       | Node affinity preset type. Ignored if `affinity` is set. Allowed values: `soft` or `hard`                                                             | `""`                                                                                                        |
+| `nodeAffinityPreset.key`                        | Node label key to match Ignored if `affinity` is set.                                                                                                 | `""`                                                                                                        |
+| `nodeAffinityPreset.values`                     | Node label values to match. Ignored if `affinity` is set.                                                                                             | `[]`                                                                                                        |
```
Comment (juan131) [inline bitnami/kong/README.md:89]: Add this section to the README.md:
```
### Setting Pod's affinity

This chart allows you to set your custom affinity using the `affinity` paremeter. Find more infomation about Pod's affinity in the [kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity).

As an alternative, you can use of the preset configurations for pod affinity, pod anti-affinity, and node affinity available at the [bitnami/common](https://github.com/bitnami/charts/tree/master/bitnami/common#affinities) chart. To do so, set the `podAffinityPreset`, `podAntiAffinityPreset`, or `nodeAffinityPreset` parameters.
```

File: bitnami/kong/templates/deployment.yaml [hunk 4]
```diff
@@ -29,55 +35,60 @@ spec:
...
       {{- if .Values.schedulerName }}
       schedulerName: {{ .Values.schedulerName | quote }}
       {{- end }}
       {{- if .Values.affinity }}
-      affinity: {{- include "kong.tplValue" (dict "value" .Values.affinity "context" $) | nindent 8 }}
+      affinity: {{- include "common.tplvalues.render" (dict "value" .Values.affinity "context" $) | nindent 8 }}
+      {{- else }}
+      affinity:
+        podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.podAffinityPreset "context" $) | nindent 10 }}
+        podAntiAffinity: {{- include "common.affinities.pods" (dict "type" .Values.podAntiAffinityPreset "context" $) | nindent 10 }}
```
Comment (juan131) [inline bitnami/kong/templates/deployment.yaml:56]:
```suggestion
        podAffinity: {{- include "common.affinities.pods" (dict "type" .Values.podAffinityPreset "component" "server" "context" $) | nindent 10 }}
        podAntiAffinity: {{- include "common.affinities.pods" (dict "type" .Values.podAntiAffinityPreset "component" "server" "context" $) | nindent 10 }}
```

**Review by juan131 — APPROVED** [review]
Approving your PR! Please implement these two suggestions before merging it

**Pull request merged**